components/color_input.tsx

```diff
@@ -169,7 +164,7 @@ class ColorInput extends React.PureComponent<Props, State> {
                         id={`${id}-ChromePickerModal`}
                     >
                         <ChromePicker
-                            color={color}
+                            color={value}
                             onChange={this.handleColorChange}
```

Comment:
Maybe only propagate at end of dragging? (debounce)

```suggestion
                            onChange={this.handleTempColorChange} // setState
                            onChangeComplete={this.handleColorChange} // props.onChange
```

Reply:
I was originally planning on something like that, but that's needed to update the "preview" theme as you drag (see the sidebar above). We could consider changing it at some point, but this matches current behaviour for now.

Reply:
I didn't realize that was required (live while dragging). 

I was testing on a local dev build, in Chrome (12+ tabs) with devtools open, and I was seeing quite a bit of lag (only applying change on mouseup). I tried tried applying this (onChange + onChangeComplete), and thats what brought up the `getDerivedStateFromProps` comment/suggestion. (When opening picker, `focused` is still true until the first drag has ended.)

The production build has much better performance—especially with devtools closed 😛.

Reply:
I think the performance issues are mostly caused by `applyTheme` being awful 😛 

Some debouncing might help that, but I'm not too concerned about it right now